operator/runner.go

```diff
@@ -26,12 +29,16 @@ import (
 // or another type. It does not support certain advanced app.App functionality which is not natively supported by
 // CRDs, such as arbitrary subresources (app.App.CallSubresource). It should be instantiated with NewRunner.
 type Runner struct {
-	config        RunnerConfig
-	webhookServer *webhookServerRunner
-	metricsServer *metricsServerRunner
-	startMux      sync.Mutex
-	running       bool
-	runningWG     sync.WaitGroup
+	config            RunnerConfig
+	webhookServer     *webhookServerRunner
+	metricsExporter   *metrics.Exporter
+	serverRunner      *app.SingletonRunner
+	healthCheck       health.HealthCheck
+	healthCheckRunner *health.HealthCheckRunner
```

Comment:
A couple of nits – these should IMHO be just one component, that implements `Run` and `AddChecks`. Also, the interface should be defined here (i.e. on the caller side), not on the implementation side, following the Go pattern of "return concrete types, use interface types".

Something like:
```go
type HealthCheck interface {
	app.Runnable
	AddChecks(...health.Check) error
}
```

Reply:
Of course! converting it to interface is definitely on my plate. Combining them makes it easier - which I had a [hint](https://github.com/grafana/grafana-app-sdk/pull/626#discussion_r1950005018) about doing, just wanted to confirm my thinking with you both.

Reply:
>the Go pattern of "return concrete types, use interface types".

Learned something new!

Reply:
Some projects I have worked with so far define the interface and struct in the same file: https://github.com/grafana/authlib/blob/bf4600a17569ead7f7ddd49ff00136916896c4ce/authn/verifier.go#L69-L78

Is that considered difficult on the eyes?

Reply:
> What if you have multiple call sites spread across various files?

You can define your imported interface in one place, I'd put it in the root package file (i.e. `package/package.go` or `package/types.go`). But in my experience this has rarely been the case.

> Some projects I have worked with so far define the interface and struct in the same file.

Returning and interface is usually done when you have a package that has multiple implementations of the same one. Still, personally I always to prefer to return concrete types from initializers and use interface compatibility checks.

https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a is a good read on why's and how's.